Регистарция пользователя и доменная модель

Vitaly-Pon opened the pull request.

Vitaly-Pon committed:
Добавил доменную сущность
Создал маппер

Vitaly-Pon committed:
Реализация регистрации пользователя
Создание конфинга для защиты

AlexanderNest commented on .idea/dataSources.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
GITIGNORE

AlexanderNest commented on .idea/encodings.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
GITIGNORE

AlexanderNest commented on core/src/main/java/com/example/demo/controller/dto/response/MovieRecommendationsResponse.java:
```diff
@@ -9,5 +9,9 @@
+    private List<MovieRecommendationModel> movies;
+
+    public static MovieRecommendationsResponse fromDomainList(List<MovieRecommendationModel> domainMovies) {
+        return new MovieRecommendationsResponse(domainMovies);
```
это не маппинг, это переупаковка

AlexanderNest commented on core/src/main/java/com/example/demo/controller/UserController.java:
```diff
@@ -0,0 +1,42 @@
+@RestController
+@RequestMapping("/api/users")
+@AllArgsConstructor
+public class UserController {
```
где интерфейс?

AlexanderNest commented on core/src/main/java/com/example/demo/controller/UserController.java:
```diff
@@ -0,0 +1,42 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/users")
```
зачем /api?

Vitaly-Pon committed:
Реализация регистрации пользователя
Создание конфинга для защиты

AlexanderNest commented on core/src/main/java/com/example/demo/deepseek/MovieResponseParser.java:
```diff
@@ -1,24 +1,39 @@
+        return wrapper.getMovies();
+    }
+    // Вложенный класс для обертки JSON
+    private static class MoviesWrapper {
```
вложенные классы не делаем

AlexanderNest commented on core/src/main/java/com/example/demo/entity/User.java:
```diff
@@ -0,0 +1,37 @@
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
```
где перенос?

AlexanderNest commented on core/src/main/java/com/example/demo/entity/User.java:
```diff
@@ -0,0 +1,37 @@
+
+    @PrePersist //автоматически вызывается перед сохранением сущности в базу.
+    protected void onCreate() {
+        this.createdAt = LocalDateTime.now();
```
для этого есть аннотация. либо вынести на сторону БД

Vitaly-Pon committed:
Реализация регистрации пользователя
Создание конфинга для защиты

AlexanderNest commented on core/src/main/java/com/example/demo/service/UserService.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public void deleteUser(Long id) {
+        if (!userRepository.existsById(id)) {
```
это зачем? по REST удаление несуществующего - это тоже ОК

AlexanderNest commented on core/src/main/java/com/example/demo/DemoApplication.java:
```diff
@@ -2,10 +2,14 @@
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
 @SpringBootApplication
+@EntityScan(basePackages = "com.example.demo.entity") // чтобы Hibernate видел сущности
```
и без этого все будет работать, если подключать spring boot starter jpa

AlexanderNest commented on core/src/main/java/com/example/demo/DemoApplication.java:
```diff
@@ -2,10 +2,14 @@
 
 @SpringBootApplication
+@EntityScan(basePackages = "com.example.demo.entity") // чтобы Hibernate видел сущности
+@EnableJpaRepositories(basePackages = "com.example.demo.repository") // чтобы Spring Data видел репозитории
```
туда же

AlexanderNest commented on core/src/main/resources/application.properties:
```diff
@@ -6,3 +6,14 @@ deepseek.api.key=${DEEPSEEK_API_KEY:d}
+spring.datasource.username=sa
+spring.datasource.password=
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.jpa.hibernate.ddl-auto=update
```
VALIDATE

AlexanderNest commented on core/src/main/java/com/example/demo/service/UserService.java:
```diff
@@ -0,0 +1,43 @@
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder; // внедрится бин
+
+    public User register(UserRegistrationRequest request) {
```
каждый слой должен возвращать свою сущность.

AlexanderNest reviewed (changes requested):
.

Vitaly-Pon committed:
Переименование классов для лучшего отражения сути
Добавление интерфейсов
Создание маппера для пользователя

Vitaly-Pon committed:
Переименование классов для лучшего отражения сути
Добавление интерфейсов
Создание маппера для пользователя

Vitaly-Pon committed:
Подключил liquibase
Создание MovieRecommendationDto
Ошибка в liquibase? при запуске

Vitaly-Pon committed:
Подключил liquibase
Создание MovieRecommendationDto
Ошибка в liquibase? при запуске

Vitaly-Pon committed:
refactoring

Vitaly-Pon committed:
ВСЕ РАБОТАЕТ

Vitaly-Pon committed:
ВСЕ РАБОТАЕТ

Vitaly-Pon committed:
Изменил структуру проекта
Добавил response и request касательно истории и рейтинга
рейтинг теперь не целое число, а 7,5
Добавил логирование
Изменил фильтрацию уже оцененных фильмов

Vitaly-Pon committed:
Изменил структуру проекта
Добавил response и request касательно истории и рейтинга
рейтинг теперь не целое число, а 7,5
Добавил логирование
Изменил фильтрацию уже оцененных фильмов

Vitaly-Pon committed:
После занятия

Vitaly-Pon committed:
Создал сущность movie и таблицу
Переместил @GetMapping("{userId}/ratings") в UserController
Очистил базу и чистый changeSet с новыми таблицами
Удалил лишние неиспользуемые реквесты

Vitaly-Pon committed:
Переделал тесты
Создал новые тесты
